src/server.ts

```diff
@@ -487,22 +494,29 @@ export class Server {
    * @param {string} [friendbotUrl] - The optional explicit address for
    *    friendbot. If not provided, the client will call the Soroban-RPC `getNetwork`
    *    method to attempt to find this network's friendbot url.
-   * @returns {Promise<boolean>} Returns a promise to a boolean value
-   *    indicating whether the account was created or already existed.
+   * @returns {Promise<Account>} Returns a promise to the {@link Account} object with populated sequence number.
    */
   public async requestAirdrop(
     address: string | Pick<Account, "accountId">,
     friendbotUrl?: string,
-  ): Promise<boolean> {
+  ): Promise<Account> {
     const account = typeof address === "string" ? address : address.accountId();
     friendbotUrl = friendbotUrl || (await this.getNetwork()).friendbotUrl;
     if (!friendbotUrl) {
       throw new Error("No friendbot URL configured for current network");
     }
-    const response = await AxiosClient.post<Friendbot.Response>(
-      `${friendbotUrl}?addr=${encodeURIComponent(account)}`
-    );
-    return !!response.data?.successful;
+    try {
+      await AxiosClient.post<Friendbot.Response>(
+        `${friendbotUrl}?addr=${encodeURIComponent(account)}`,
+      );
+      return new Account(account, "1");
+    } catch (error) {
+      if (error.response?.status === 400) {
+        // Account already exists, load the sequence number
+        return this.getAccount(account);
+      }
+      throw error;
+    }
```

Comment:
Imma be annoying for a sec for my own edification. I know you mentioned preference for try/catch in the dApp, but are we / should we do a similar thing here? what about chaining?

```ts
return AxiosClient
    .post<Friendbot.Response>(`...`)
    .then((account) => new Account(account, account.sequence)
    .catch((err) => {
        if (err.response?.status === 400) {
            return this.getAccount(account);
        }
        return err;
    });
```

Reply:
Also, note two additional differences in the above:

 - there's no need to `await` the friendbot request since we return a Promise, anyway (right?)
 - the sequence number isn't necessarily 1 after funding so we need to check the `post` retval (I think?)

Reply:
(Can you add a test for the second bullet point, i.e. ensure the friendbot-funded return value matches the getAccount return value?)

Reply:
Why wouldn't the sequence number on a newly created account be 1?

Reply:
and re: chaining. We haven't needed to do either here, yet. Generally I've been using async/await. But whatever you prefer.

Reply:
> Why wouldn't the sequence number on a newly created account be 1?

New account sequence number is determined by the sequence number of the ledger in which it's created (IIRC 32 bits left shifted)

Reply:
I didn't know the details, I just knew it wasn't 1. Thanks Tomer! (tl;dr: this avoids `MergeAccountOp` wonkery)

With that in mind, I do think we have to use the return value from `getAccount` to build the return value, and since it's async, `.then()`ifying it also makes sense to make the method return asap (whether or not that's done via `try`/`catch` or `.error()` is up to you; I was just asking to learn).

Reply:
Cool, TIL.